src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.FeatureCodec;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.stream.StreamSupport;
+
+/**
+ * General Feature walker.
+ * <p>
+ *     For tools that iterate along a list of features.
+ * </p>
```

Comment:
Improve class-level docs:

 * A FeatureWalker is a tool that processes a Feature at a time from a source of Features, with
 * optional contextual information from a reference, sets of reads, and/or supplementary sources
 * of Features.
 *
 * FeatureWalker authors must implement the apply method to process each Feature, as well as isAcceptableFeatureType and getDrivingFeatureFile, and may optionally implement
 * onTraversalStart, onTraversalSuccess, and/or closeTool